linkerd/proxy/http/src/h1.rs

```diff
@@ -108,9 +117,17 @@ where
 
             if client.is_none() {
                 debug!(use_absolute_form, "Caching new client");
-                *client = Some(hyper::Client::builder().set_host(use_absolute_form).build(
-                    HyperConnect::new(self.connect.clone(), self.target.clone(), use_absolute_form),
-                ));
+                *client = Some(
+                    hyper::Client::builder()
+                        .pool_max_idle_per_host(self.pool.max_idle)
+                        .pool_idle_timeout(self.pool.idle_timeout)
```

Comment:
here's a thought, take it or leave it: I wonder if it would make sense to just pass around a builder, rather than a `PoolSettings`? may not be the right approach...